source/Classes/020.DnsRecordCname.ps1

```diff
@@ -25,6 +27,14 @@ class DnsRecordCname : DnsRecordBase
 
     DnsRecordCname()
     {
+        # Per RFC 1035 DNS standards, this regex ensures valid FQDN format requiring a terminating dot (based on AI verification)
+        $HostNameAliasRegex = '^(?!.{254})(?:(?!-)[a-z0-9-]{1,63}(?<!-)\.)+(?:(?!-)[a-z0-9-]{1,63}(?<!-))\.$'
```

Comment:
We need tests to verify this regex when it fails - all possible combinations to verify that they fail as expected. Also tests that verify passing names that should pass, multiple longest domain names and sub domains.